trio/_wait_for_single_object.py

```diff
@@ -0,0 +1,56 @@
+from . import _timeouts
+from . import _core
+from ._threads import run_sync_in_worker_thread
+from ._core._windows_cffi import ffi, kernel32, ErrorCodes
+
+
+class StubLimiter:
+    def release_on_behalf_of(self, x):
+        pass
+
+    async def acquire_on_behalf_of(self, x):
+        pass
+
+
+async def WaitForSingleObject(handle):
+    """Async and cancellable variant of kernel32.WaitForSingleObject().
+
+    Args:
+      handle: A win32 handle.
+
+    """
+    # Quick check; we might not even need to spawn a thread. The zero
+    # means a zero timeout; this call never blocks. We also exit here
+    # if the handle is already closed for some reason.
+    retcode = kernel32.WaitForSingleObject(handle, 0)
```

Comment:
This should have an `if retcode == ErrorCodes.WAIT_FAILED: raise_winerror()` branch too, I think.

Reply:
Good point. I initially though that just returning on closed handled would be practical, but if someone somehow passes a wrong/invalid handle, they'd want to know.

Reply:
Yeah... I was also confused about this until I thought about it more, and remembered that there isn't really such a thing as a "closed handle" – once you close a handle, the next new handle you open has a chance of being assigned the same handle id. So doing anything with a handle after you close it is really error prone – you might end up doing it on some other random object without realizing it! Better to get an error early in testing and fix the bug.